packages/components/dialog/src/dialog.ts

```diff
@@ -94,6 +97,14 @@ export class Dialog extends ScopedElementsMixin(LitElement) {
     super.connectedCallback();
 
     this.inert = true;
+
+    this.#focusTrapController = new FocusTrapController(this, { disableCancel: this.disableCancel });
+  }
+
+  override disconnectedCallback(): void {
+    super.disconnectedCallback();
+
+    this.#focusTrapController?.deactivate();
```

Comment:
Can't this happen automatically? In the `hostDisconnected` callback in the controller?